### PR TITLE
VMO-4144 Unified Open Block

### PR DESCRIPTION
**Pull request by poojahpatil90**
https://user-images.githubusercontent.com/23651142/126882220-d7ad3970-7aef-4cee-bcd2-c4343962db1d.mov

### CODE REVIEW

File: src/components/interaction-designer/block-types/MobilePrimitives_OpenResponseBlock.vue [hunk 1]
```diff
@@ -13,6 +13,8 @@
         :block="block" />
       <block-name-editor :block="block" />
 
+      <hr>
```
Comment (safydy-r) [inline src/components/interaction-designer/block-types/MobilePrimitives_OpenResponseBlock.vue:16]: I don't think we need `<hr>` before `block-max-duration-seconds-editor`, right ?

File: src/components/interaction-designer/block-types/MobilePrimitives_OpenResponseBlock.vue [hunk 2]
```diff
@@ -66,6 +74,7 @@ import BlockId from '../block-editors/BlockId.vue'
 import BlockMaxDurationSecondsEditor from '../block-editors/MaxDurationSecondsEditor.vue'
 import BlockMaxResponseCharactersEditor from '../block-editors/MaxResponseCharactersEditor.vue'
 import GenericContactPropertyEditor from '../block-editors/GenericContactPropertyEditor.vue'
+import BlockOutputBranchingConfig from '@/components/interaction-designer/block-editors/BlockOutputBranchingConfig.vue'
```
[the remark this answers is not in the document]
Comment (safydy-r) [reply src/components/interaction-designer/block-types/MobilePrimitives_OpenResponseBlock.vue:77]: Would be great if you can take the opportunity for correct import order here as well ;)

**Review by safydy-r — CHANGES_REQUESTED** [review]
I'd be glad to approve after you made quick changes related to my comments.

File: src/components/interaction-designer/block-editors/MaxResponseCharactersEditor.vue [hunk 3]
```diff
@@ -1,21 +1,28 @@
...
-        :valid-state="isValid" />
-      <small class="text-muted">
-        {{ 'flow-builder.unlimited-if-not-defined-or-set-as-zero' | trans }}
-      </small>
-    </div>
-  </validation-message>
+  <div>
+    <span class="d-flex mb-2">
+      <input type="checkbox" class="align-self-center" v-model="showMaxResponseCharField" />
+      <p class="m-0 ml-2">Include maximum characters to prompt for</p>
```
[the remark this answers is not in the document]
Comment (poojahpatil90) [reply src/components/interaction-designer/block-editors/MaxResponseCharactersEditor.vue:5]: Good catch!  I fixed it

File: src/components/interaction-designer/block-editors/MaxResponseCharactersEditor.vue [hunk 4]
```diff
@@ -44,6 +51,7 @@ export default {
   data() {
     return {
       defaultMaxLength: 0,
+      showMaxResponseCharField: false
```
[the remark this answers is not in the document]
Comment (poojahpatil90) [reply src/components/interaction-designer/block-editors/MaxResponseCharactersEditor.vue:54]: Done

File: src/components/interaction-designer/block-types/MobilePrimitives_OpenResponseBlock.vue (hunk 1 above)
Comment (poojahpatil90) [reply src/components/interaction-designer/block-types/MobilePrimitives_OpenResponseBlock.vue:16]: Screen Shot 2021-07-26 at 4 20 08 PM

@safydy-r  As seen in this Design screenshot (DSN-111). It has a divider between the title and rest of the block fields.

Comment (safydy-r) [reply src/components/interaction-designer/block-types/MobilePrimitives_OpenResponseBlock.vue:16]: I see, you're right. I might have been confused somewhere :D

**Review by safydy-r — APPROVED** [review]
Nice one @poojahpatil90

**Pull request merged**